lib/chatty/connection.ex

```diff
@@ -119,10 +144,24 @@ defmodule Chatty.Connection do
     {:noreply, %{state | sock: nil, last_message_time: nil}}
   end
 
+  def handle_info({:ssl_error, sock, reason}, %{sock: sock} = state) do
```

Comment:
This duplication also needs to be removed. I would add `irc_close` to the helpers and call it here instead of `:gen_tcp.close(sock)` or `:ssl.close(sock)`. That way we can keep the conditionals that check `@ssl` in mostly a single place.